internal/eval/evalers.go

```diff
@@ -1073,3 +1120,123 @@ func (n *ipIsInRangeEval) Eval(ctx *Context) (types.Value, error) {
 	}
 	return types.Boolean(rhs.Contains(lhs)), nil
 }
+
+// virtualLessThanEval struct
+type virtualLessThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualLessThanEval(lhs Evaler, rhs Evaler) *virtualLessThanEval {
+	return &virtualLessThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualLessThanEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(lhs.Less(rhs)), nil
+}
+
+var _ Evaler = &virtualLessThanEval{}
+
+// virtualGreaterThanEval struct
+type virtualGreaterThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualGreaterThanEval(lhs Evaler, rhs Evaler) *virtualGreaterThanEval {
+	return &virtualGreaterThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualGreaterThanEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(!lhs.LessEqual(rhs)), nil
+}
+
+// virtualLessEqualThanEval struct
+type virtualLessThanOrEqualEval struct {
```

Comment:
I'd rather it be

`lesserLessThanOrEqualEval`

(or if we renamed to Comparable)

`comparableLessThanOrEqualEval`

Reply:
I'll wait on this until the comparison stuff settles.